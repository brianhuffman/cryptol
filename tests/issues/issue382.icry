0b01 : Bit
(True, False, True) : (Bit, Bit)
0b01010 : [4]
(True, False) : Bit
{a = True} : {a : Bit, b : Bit}
True : [2]
[] : [100]
0b010 + {}
0b010 + (True, False)
(\(x:Bit) -> x) : [2]
True True : Bit
