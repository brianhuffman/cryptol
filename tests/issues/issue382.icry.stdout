Loading module Cryptol

[error] at issue382.icry:1:1--1:5:
  Type mismatch:
    Expected type: Bit
    Inferred type: [2]
    When checking user annotation

[error] at issue382.icry:2:1--2:20:
  Type mismatch:
    Expected type: (Bit, Bit)
    Inferred type: (Bit, Bit, Bit)
    When checking user annotation

[error] at issue382.icry:3:1--3:8:
  Type mismatch:
    Expected type: 4
    Inferred type: 5
    Context: [ERROR] _
    When checking user annotation

[error] at issue382.icry:4:1--4:14:
  Type mismatch:
    Expected type: Bit
    Inferred type: (Bit, Bit)
    When checking user annotation

[error] at issue382.icry:5:1--5:11:
  Type mismatch:
    Expected type: {a : Bit, b : Bit}
    Inferred type: {a : Bit}
    Missing field b
    When checking user annotation

[error] at issue382.icry:6:1--6:5:
  Type mismatch:
    Expected type: [2]
    Inferred type: Bit
    When checking user annotation

[error] at issue382.icry:7:1--7:3:
  Type mismatch:
    Expected type: 100
    Inferred type: 0
    When checking length of sequence

[error] at issue382.icry:8:9--8:11:
  Type mismatch:
    Expected type: [3]
    Inferred type: {}
    When checking type of function argument

[error] at issue382.icry:9:9--9:22:
  Type mismatch:
    Expected type: [3]
    Inferred type: (Bit, Bit)
    When checking type of function argument

[error] at issue382.icry:10:2--10:15:
  Type mismatch:
    Expected type: [2]
    Inferred type: Bit -> Bit
    When checking user annotation

[error] at issue382.icry:11:1--11:5:
  Type mismatch:
    Expected type: Bit -> Bit
    Inferred type: Bit
    When checking function call
